src/Modules/UngradedModules/ModuleGens.jl

```diff
@@ -283,18 +283,21 @@ function (F::FreeMod{<:MPolyRingElem})(s::Singular.svector)
   return FreeModElem(sparse_row(Rx, [(i, finish(ctx)) for (i, ctx) in cache]), F)
 end
 
+# TODO: move this eventually
+length(s::Singular.svector) = Int(Singular.libSingular.pLength(s.ptr))
```

Comment:
Please move this to Singular.jl before merging this PR. Sorting this out afterwards is more of a hassle than doing it right the first time